docker/init_scripts/init

```diff
@@ -173,38 +174,54 @@ start_bin_scheduler() {
 		RQ_REDIS_SSL=${REDIS_SSL:-0} \
 		rqscheduler \
 		--path /backend \
-		--pid /tmp/scheduler.pid &
+		--pid /tmp/rq_scheduler.pid &
 }
 
-# function that runs our independent python scripts and creates corresponding PID files,
-start_python() {
-	SCRIPT="${1}"
-	info_log "Starting ${SCRIPT}"
-	python3 "${SCRIPT}.py" &
+# Commands to start RQ worker
+start_bin_rq_worker() {
+	info_log "Starting rq worker"
+
+	# Build Redis URL properly
+	local redis_url
+	if [[ -n ${REDIS_USERNAME-} && -n ${REDIS_PASSWORD-} ]]; then
+		redis_url="redis${REDIS_SSL:+s}://${REDIS_USERNAME}:${REDIS_PASSWORD}@${REDIS_HOST:-127.0.0.1}:${REDIS_PORT:-6379}/${REDIS_DB:-0}"
+	elif [[ -n ${REDIS_PASSWORD-} ]]; then
+		redis_url="redis${REDIS_SSL:+s}://:${REDIS_PASSWORD}@${REDIS_HOST:-127.0.0.1}:${REDIS_PORT:-6379}/${REDIS_DB:-0}"
```

Comment:
We are not correctly handling the scenario when there's a user but no password. The simple approach for this would be:

1. If there's a password: Use `${REDIS_USERNAME}:${REDIS_PASSWORD}`, so even if there's no user, the auth will correctly set `:<password>`.
2. If there's a user: Use `${REDIS_USERNAME}`. We need to avoid adding a trailing `:` in this scenario.
2. Fallback to scenario without Basic authentication.